### DISCUSSION TITLE
Dynamic import progress

### DISCUSSION THREAD

**Discussion by zignis** [vercel/next.js#40901] [Help] [answered]
Can we somehow compute the import progress of dynamically loaded components? 

### Idea
```tsx
import dynamic from 'next/dynamic';

const MyComp = dynamic(() => import('modules/mycomp'), {
  loading: ({ progress }: { progress: number }) => <div>Loading... {progress}%</div>,
});
```

<hr />

*Update:* Seems like this is not possible yet as next uses `react.lazy`, which is based on promises so it's either fulfilled or rejected. Although we can fake the progress using something like `nprogress`.

**Comment by busy-dog** [comment]
I have the same problem

**Reply by zignis** [reply, marked as answer]
Sorry but there's no solution to it except faking the progress. Should you want this feature implemented, file a new issue or discussion in the react repo, as this would require rewriting `react.lazy`.
